### ISSUE TITLE
Fix executable permission for the maven wrapper before invoking the wrapper

### ISSUE THREAD

**Issue by ArieShout** [paketo-buildpacks/maven#6]
If we archive the source code to a tar.gz ball in Windows, it's likely we will lost the execute permission bit of the Maven wrapper (mvnw).

When the maven buildpack starts the build, it checks the existence of the maven wrapper `mvnw`, but it does not check the executable permission. During the build, we see the following error:

```
Pivotal Build System Buildpack 1.0.97
  fork/exec /workspace/mvnw: permission denied
```

We may need either

* skip mvnw if it's not marked as executable
* or, set the execution permission of the file before we start the build.

**Comment by nebhale** [comment]
@ArieShout are you worried about the permissions on any of the other files?

**Comment by ArieShout** [comment]
Hi @nebhale , we didn't notice other file permission issue apart from this.
Thanks for the update!

**Comment by agracey** [comment]
@nebhale I think we might be seeing a similar issue in our usage with Epinio. 

```
error running build
unable to start PTY
fork/exec /workspace/source/app/mvnw: no such file or directory
```

From everything that I can see, the file does indeed exist at that location and when I delete the mvnw files, it starts building correctly. Is this potentially the same bug? (I'm using the 5.3.2 build pack from last week)

**Comment by dmikusa-pivotal** [comment]
@agracey If you would be so kind, can you open a new issue & include some steps to reproduce this? When you get that information posted, I can take a look and see what we can do.

**Comment by dtillman** [comment]
@dmikusa-pivotal : Daniel, I can confirm the above error!.  I did a clean install of everything (like a new user just trying out CNB for first time).  Unfortunately the experience was not great as the sample build fails.

I used the CNB sample from "https://github.com/paketo-buildpacks/samples/java/maven" and it will not build on a windows 10 machine with Docker Desktop (latest) using WSL2 as backend.   I'm using the latest pack (0.19.0+git-360dbae.build-2550)  with pack config default-builder paketobuildpacks/builder:base .... 

After googling for the error I found this thread and I deleted mvnw and the build completed!

Dave
